### ISSUE TITLE
[a11y]: DatePicker range variety - clears start date after tabbing off end date

### ISSUE THREAD

**Issue by tombrunet** [carbon-design-system/carbon#12734]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.17.0

### Automated testing tool and ruleset

Manual

### Assistive technology

Keyboard

### Description

Typing an end date and tabbing off of the field clears the start date

### Reproduction/example

Storybook

### Steps to reproduce

See https://react.carbondesignsystem.com/?path=/docs/components-datepicker--range-with-calendar
* Tab to the Range Datepicker 
* Type a valid date into the "Start Date"
* Press tab (focus now moves to the "End Date" and keeps the start date)
* Type a valid date into the "End Date"
* Press tab (focus now moves off the widget. The "End Date" is kept, but the "Start Date" now clears)

Note - this does not happen if you first set dates using the mouse

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by francinelucca** [comment]
I was unable to reproduce the bug as described, but I do see a weird behavior on the *end date* being removed after tabbing off, tabbing back and then tabbing in again, I'll take a look into this behavior and do further testing

https://user-images.githubusercontent.com/40550942/216638883-29aff828-f6ff-4a8f-8329-275198c2b807.mov
